lib/vmpooler/api/v1.rb

```diff
@@ -509,6 +514,47 @@ def need_token!
 
       JSON.pretty_generate(result)
     end
+
+    post "#{api_prefix}/vm/:hostname/snapshot/?" do
+      content_type :json
+
+      status 404
+      result = { 'ok' => false }
+
+      params[:hostname] = hostname_shorten(params[:hostname], config['domain'])
+
+      if backend.exists('vmpooler__vm__' + params[:hostname])
+        result[params[:hostname]] = {}
+
+        o = [('a'..'z'), ('0'..'9')].map(&:to_a).flatten
+        result[params[:hostname]]['snapshot'] = o[rand(25)] + (0...31).map { o[rand(o.length)] }.join
```

Comment:
I agree with the usefulness of a `create_a_random_string_of_N_characters` method.  Currenlty both `pool_manager` and `api/v1` have randomization code in them, and unfortunately there isn't currently a helper library which is shared throughout the app.  I can certainly refactor some code to accomplish this, but am unsure if it's out of the scope of the PR.